### ISSUE TITLE
[a11y]: Modal: Most of the modals are not properly dismissable in storybook

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#13710]
Originally outlined in #12727:

> The modals are not launchable or dismissable in storybook, except the one "With State Manager". This means basic modal behaviour cannot be assessed. This just seems like an implementation bug, but since modals have very specific behaviour, it's important the demos are functional
